setup.sh

```diff
@@ -14,29 +14,34 @@ normalize_version() {
 }
 
 latest_version() {
-  jq --arg channel "$1" '.releases | map(select(.channel==$channel)) | first'
+  jq --arg channel "$1" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.channel==$channel) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'
```

Comment:
I think we can use something like `select(.dart_sdk_arch as $sdk_arch | $arch | index($sdk_arch))` with argument like:

- `--argjson arch '["x64", null]'` for x64 and default value, or
- `--argjson arch '["arm64"]'` when looking for arm64